[TEST MERGE] Slot-based Storage

EmoGarbage404 opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
Fixes #21134

Storage now functions on a slot-based system instead of a numerical size one.

Storages now specify an amount of items and an optional max size and optional max total weight.
Leaving the max size or max total weight null simply fills it in with appropriate defaults.

Sizes are defined with an enum, and carry an additional "weight cost" which is just used internally to stop players from stuffing bags with tons of bulky items.

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
all of it is in the issue.

Most of my argument is articulated here: https://github.com/space-wizards/space-station-14/issues/21134#issuecomment-1773808371

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->
ItemComponent's Size is now an enum rather than an int.

In StorageComponent, Capacity has been replaced with MaxSlots, MaxItemSize (nullable) and MaxTotalWeight (nullable)

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
![image](https://github.com/space-wizards/space-station-14/assets/98561806/486f1869-56ce-4cab-a99a-321846ee1767)
![image](https://github.com/space-wizards/space-station-14/assets/98561806/bd09aa42-2842-4b5d-b572-ee9c9d89a396)

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->
Some ItemSystem functions take in the ItemSize enum rather than a simple int.

A few storageSystem functions now use Entity<T> arguments in the header.

Removed Capacity, UsedStorage vars from StorageComponent.

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

:cl:
- tweak: Storage now functions based on slots and sizes, rather than numerical values.

DogZeroX commented:
Sloth based

LankLTE commented:
This is inevitably going to cause a ton of discussion so I’ll just add: I don’t think the issue that you said “contains everything” properly justifies why the entire inventory system needs to be changed into a system so vastly different and arguably more restrictive than the one we currently have. Evidently looking at the issue I’m not the only one who feels this way- I know oftentimes votes / reactions are brushed off but I do think this is a situation where the overwhelmingly negative feedback from players is important to look at rather than just.. doing it anyways.

LordEclipse commented:
So this is basically only really being created because of medical items.

EmoGarbage404 commented:
> So this is basically only really being created because of medical items.

yeah i mean if you ignore everything i've said in the entire lengthy discussion, you could come to that opinion.

keronshb commented:
![image](https://github.com/space-wizards/space-station-14/assets/54602815/2964a2fb-e5f7-4766-bb5a-1dc18aeea1f8)

As a long time Engi/Atmos player, this is satisfactory in comparison to SS13.

![image](https://github.com/space-wizards/space-station-14/assets/54602815/10122089-2f54-4e42-bc17-1be4c3a0ca1f)

LordEclipse commented:
> > So this is basically only really being created because of medical items.
> 
> yeah i mean if you ignore everything i've said in the entire lengthy discussion, you could come to that opinion.

You only made those to argue for it, the original issue was because of medical item sizes.

metalgearsloth commented:
https://github.com/space-wizards/space-station-14/issues/21134#issuecomment-1773808371

I think emo's argument is pretty well articulated here so I think it's disingenuous to say it's because of med.

LordEclipse commented:
> [#21134 (comment)](https://github.com/space-wizards/space-station-14/issues/21134#issuecomment-1773808371)
> 
> I think emo's argument is pretty well articulated here so I think it's disingenuous to say it's because of med.

It has nothing to do with medical, its the entire changing of inventory management. I said that the original issue was only started because of medical items, and Emo argued the benefits of slot-based inventory. Plus, some systems or items being subject to some vague concept of realism is spotty across the board. Is realism wanted for most things? Realism isn't always the best indicator of a good game mechanic.

keronshb commented:
> > [#21134 (comment)](https://github.com/space-wizards/space-station-14/issues/21134#issuecomment-1773808371)
> > I think emo's argument is pretty well articulated here so I think it's disingenuous to say it's because of med.
> 
> It has nothing to do with medical, its the entire changing of inventory management. I said that the original issue was only started because of medical items, and Emo argued the benefits of slot-based inventory. Plus, some systems or items being subject to some vague concept of realism is spotty across the board. Is realism wanted for most things? Realism isn't always the best indicator of a good game mechanic.

How many items do you realistically need in your backpack?

I don't even have my storage optimized and I can carry _a lot_ as an engineer. About what's required of me + more. It also matches SS13's system to an extent. And I can carry an entire armory in that game in my bag.

LankLTE commented:
Honestly I also don't like the idea of just nerfing storage levels across the board from a gameplay standpoint. I have a lot of issues with my PR but I think that's a big one that should be addressed with something other than "13 does it".

asperger-sind commented:
> Honestly I also don't like the idea of just nerfing storage levels across the board from a gameplay standpoint. I have a lot of issues with the PR but I think that's a big one that should be addressed with something other than "13 does it".

Yeah, big storage is fun, it removes tedium where it could be and while it may raise balance concerns at times I think it's better to have it anyway.

keronshb commented:
> Honestly I also don't like the idea of just nerfing storage levels across the board from a gameplay standpoint. I have a lot of issues with my PR but I think that's a big one that should be addressed with something other than "13 does it".

You don't realize how much SS13 players have optimized storage. I can powergame meds easily if I wanted to while still carrying items. Why do you need to carry way more than you need to?

This can open the road to more interesting items in the future like Durathread Backpacks & other items that allow for more/larger storage. Or alternate ideas to storage.

LordEclipse commented:
> > > [#21134 (comment)](https://github.com/space-wizards/space-station-14/issues/21134#issuecomment-1773808371)
> > > I think emo's argument is pretty well articulated here so I think it's disingenuous to say it's because of med.
> > 
> > 
> > It has nothing to do with medical, its the entire changing of inventory management. I said that the original issue was only started because of medical items, and Emo argued the benefits of slot-based inventory. Plus, some systems or items being subject to some vague concept of realism is spotty across the board. Is realism wanted for most things? Realism isn't always the best indicator of a good game mechanic.
> 
> How many items do you realistically need in your backpack?
> 
> I don't even have my storage optimized and I can carry _a lot_ as an engineer. About what's required of me + more. It also matches SS13's system to an extent. And I can carry an entire armory in that game in my bag.

From an entirely personal perspective but: I don't find joy in having to micromanage nested inventories just to make the most optimal usage of my inventory.

LankLTE commented:
> You don't realize how much SS13 players have optimized storage. I can powergame meds easily if I wanted to while still carrying items. Why do you need to carry way more than you need to?

Being able to carry exactly as much as I need isn't fun from my point of view, honestly. It just means I'm spending more time running around and grabbing stuff. It's also not really fixed by this PR as anybody who isn't an engineer carrying around 30 stack of steel, glass and rods still have room to "powergame" with things.

> This can open the road to more interesting items in the future like Durathread Backpacks & other items that allow for more/larger storage. Or alternate ideas for storage upgrades.

The current system absolutely allows for this still.

deltanedas commented:
why allow putting 7 survival boxes in a bag to get 49 items thats just annoying

metalgearsloth commented:
> why allow putting 7 survival boxes in a bag to get 49 items thats just annoying

Surv boxes are small restricted in 13 iirc, so best you're getting are emergency tanks or pills or something and not wrenches or the likes.

LankLTE commented:
> Surv boxes are small restricted in 13 iirc, so best you're getting are emergency tanks or pills or something and not wrenches or the likes.

except pretty much all of the normal tools are small so you literally can do that

EmoGarbage404 commented:
> except pretty much all of the normal tools are small so you literally can do that

here gimme 2 seconds and i'll nerf them

keronshb commented:
> > why allow putting 7 survival boxes in a bag to get 49 items thats just annoying
> 
> Surv boxes are small restricted in 13 iirc, so best you're getting are emergency tanks or pills or something and not wrenches or the likes.

You can put wrenches & tools in your boxes in SS13. It's an ok alternative to a toolbelt if you're desparate

deltanedas commented:
for the example of gas analyzer and holofan, both are useful small items
why would putting small items in a bulky box let you fit more items

LankLTE commented:
overall the point we're making here is that putting storage in storage shouldn't give you more storage- that's not how the world works and it's going to be a balance nightmare (which this PR is apparently supposed to help with)

EmoGarbage404 commented:
> overall the point we're making here is that putting storage in storage shouldn't give you more storage- that's not how the world works and it's going to be a balance nightmare (which this PR is apparently supposed to help with)

it encourages organization and eventually should be paired with blocking opening overly nested storage and potentially limited storage windows. both are a little out of scope.

keronshb commented:
> overall the point we're making here is that putting storage in storage shouldn't give you more storage- that's not how the world works and it's going to be a balance nightmare (which this PR is apparently supposed to help with)

Using nested storage is intended. If you want to have X amount of pills, put them into a pill container, then put that in your box or in your pocket.

It doesn't matter if it's how the world works, it's a game. Tetris storage in RE4 isn't how the world works but also a lot of people love that inventory system.

LankLTE commented:
> It doesn't matter if it's how the world works, it's a game.

One of Emo's big arguments was that item sizes "don't make any sense", and having to balance what we "think something weighs and the gameplay experience of how much something should weigh" is a "poor abstraction". So I think I'm allowed to point out some ways that the new system doesn't make sense either.

EmoGarbage404 commented:
> One of Emo's big arguments was that item sizes "don't make any sense", and having to balance what we "think something weighs and the gameplay experience of how much something should weigh" is a "poor abstraction". So I think I'm allowed to point out some ways that the new system doesn't make sense either.

my argument was that numerical sizes don't make sense as a way of showing the size of the item. My response to that was to use the ItemSize enum to abstract that in a more sensible way. It has no relation to the slots available.

Kindly read my argument rather than putting words in my mouth.

LankLTE commented:
> "If you want to have X amount of pills, put them into a pill container, then put that in your box or in your pocket."

Current system already supports this while also not letting you carry 49 stacks (assuming I'm going over this correctly) of bruise packs on you because you decided you wanted 7 boxes in your inventory.

LankLTE commented:
> my argument was that numerical sizes don't make sense as a way of showing the size of the item. My response to that was to use the ItemSize enum to abstract that in a more sensible way. It has no relation to the slots available.
> 
> Kindly read my argument rather than putting words in my mouth.

I did try and read over the argument and not put words in your mouth. You claimed that "size is a gamey feature", I think this is pretty in-line with the counterargument I made so apologies if it wasn't.

Aisu9 commented:
wont this change make the game even more gamey?
like i'll be able to store a singular makeshift spear, or 7 pen in my entire backpack

metalgearsloth commented:
> wont this change make the game even more gamey? like i'll be able to store a singular makeshift spear, or 7 pen in my entire backpack

That's master, now you get 7 wrenches or 7 pens if you're using normal storage on small items (I think in 13 spear is large so doesn't fit in normal storage).

Aisu9 commented:
> > wont this change make the game even more gamey? like i'll be able to store a singular makeshift spear, or 7 pen in my entire backpack
> 
> That's master, now you get 7 wrenches or 7 pens if you're using normal storage on small items (I think in 13 spear is large so doesn't fit in normal storage).

and each full stacks of medical item will be only 1 right?(doesn't matter if it's 1 or 7 or 10) only reduce the total amount of medical items on you by 30% (now we must not use decimal allowing some leeway)
Maybe we should move away from finite items and use rechargeable tools like on goon

NancokPS2 commented:
So if i am getting this right, anything that used to be size 5 now takes up an entire slot? so anything smaller than size 5 simply cannot be represented? Carrying 7 pens is the same as carrying 7 flashes?  
  
According to your arguments:  
  
> Numerical item weight doesn't correspond to actual properties of items.  
Neither does this, it is even worse in that regard than before. It IS necessary to consider how many can go on a given container, why would that not be the case?  
  
> Having storage mainly controlled by a value representing both weight and volume makes it difficult to decide capacity.    
And how does this help with that? It simply replaces issues regarding the problem with its own. Now instead of deciding weight in a modular range of 1 to 80, you have a range that jumps in (equivalent) increments of 5.    
  
> Mass whitelists for storage containers is heavy for long-term maintenance and is prone to frequent breakage  
And you're planning to not have white lists? To just allow any item as long as there's slots available? this is not just going to break stuff, its going to make it impossible to fix. If you do plan to have white lists, then this does not fix the problem.  
  

This is just trying so hard to fix a non issue, the game does not need to be this level of balanced, let alone sacrifice complexity in such a way for it. I think y'all are forgetting what the game is about.

Interrobang01 commented:
you could also fit 20 pens in an o2 locker. People probably don't need more than 14 items but if they do lockers will be a very appealing option, which is kind of cool but everyone dragging o2 lockers around probably wouldn't be very thematic

keronshb commented:
> And you're planning to not have white lists? To just allow any item as long as there's slots available? this is not just going to break stuff, its going to make it impossible to fix. If you do plan to have white lists, then this is a non-argument.

You misread it.

No mass whitelists, not no whitelists at all.

You still can't shove a hardsuit into a bag or something too huge (a fireaxe for example).

OctoRocket commented:
I really don’t want to lose the quality of life :/

metalgearsloth commented:
@PursuitInAshes what are your proposals to fix the current code burdens as you largely addressed why you were against any changes without addressing the problems with master.

keronshb commented:
> I know this might be a "minor" argument, but this game is not space station 13, It has its own identity, playerbase, community, needs. Just because 13 doesn't shouldn't mean we should, it also means 13s "balancing" shouldn't be the first thing we look at for inspiration either. 13 does some things very well, but not everything is perfect or great there. Their inventory is not something we should carry over.

This is an empty argument IMO.

While yes, we're our own game, we also aren't different.

There are mistakes from the past (SS13) that were already solved (SS13) so why repeat history?

Having a ton of limitless storage isn't a good idea for gameplay. You want enough for sure! But you don't want TOO much.

The Bag of Holding is a good example. It gives you basically the limitless storage you're looking for. But how do you balance it? Require an anomaly core? Too RNG. Limit some storage? It work: IE not being able to carry x amount of guns or ammo in your bag, which is what we implemented on Cit to prevent egrigous One Man Armys, despite still being able to OMA in some cases.

LankLTE commented:
I honestly don't see how the current code is such a burden compared to this.

Eden077 commented:
Currently from what I'm seeing and asking around with various people, this change is being recieved very negatively by many in the community, I don't understand Why you would ever change storage so drastically. If the large majority is content and happy with our current system it's probably best to currently leave it as it is.

metalgearsloth commented:
> Currently from what I'm seeing and asking around with various people, this change is being recieved very negatively by many in the community, I don't understand Why you would ever change storage so drastically. If the large majority is content and happy with our current system it's probably best to currently leave it as it is.

Coders are not happy with its current state but no solutions are being provided.

LankLTE commented:
> Coders are not happy with its current state but no solutions are being provided.

It just seems like a lot of the concerns are related to balance, which are absolutely solvable with the current system. Least that's most of the concerns that have been raised to me rather than actual coding-related issues that this system resolves.

Interrobang01 commented:
Why are coders unhappy and what solution does this PR provide? Maybe I'm not looking hard enough but past discussion doesn't seem to outline this clearly.

As far as I can tell the maintenance problems are there's too many numbers to choose from when deciding item size, and something about whitelists. What exactly are the problems, and is there a way they can they be solved without everyone hating it?

keronshb commented:
![image](https://github.com/space-wizards/space-station-14/assets/54602815/0064d971-522f-4d50-9973-2284520567f3)

I cannot understand how people are complaining. Even trying to peak powergame I find it hard to completely fill this storage.

LankLTE commented:
Because I like carrying a lot of things unrelated to my job, for fun. That's not powergaming and it's something that'll be severely impacted by this. You're suddenly forced to stop carrying around other stuff for fun and now you *have* to only carry the optimal / meta stuff because you don't comfortably have room for anything else.

keronshb commented:
> Because I like carrying a lot of things unrelated to my job, for fun. That's not powergaming and it's something that'll be severely impacted by this. You're suddenly forced to stop carrying around other stuff for fun and now you _have_ to only carry the optimal / meta stuff because you don't comfortably have room for anything else.

Please list them off to me,

Nairodian commented:
> > Because I like carrying a lot of things unrelated to my job, for fun. That's not powergaming and it's something that'll be severely impacted by this. You're suddenly forced to stop carrying around other stuff for fun and now you _have_ to only carry the optimal / meta stuff because you don't comfortably have room for anything else.
> 
> Please list them off to me,

Maybe I want to carry a mouse around for fun, or a plushie for the fuck of it, maybe an instrument on hand, a sheet of paper or two, an empty glass, put my beret in my bag when doing external work, a pAI because there's not a lot of places to put them otherwise, pack of cigs, a lighter, any of this and more, without having to replace one of my now very few, limited, and valuable item slots.

LankLTE commented:
> Please list them off to me,

Sure, I'd be glad to. 
Cigarettes.
Lighters.
Drinks.
Flashlights.
Plushies. 
Figurines.
PAI.
Instruments.
Soap.
Books.
Paper.
Mice.
Flowers.
Silk.
Hats.
Suits.
Jackets.

Just to name a few.

keronshb commented:
> > Please list them off to me,
> 
> Sure, I'd be glad to. Cigarettes. Lighters. Drinks. Flashlights. Plushies. Figurines. PAI. Instruments. Soap. Books. Paper. Mice. Flowers. Silk. Hats. Suits. Jackets.
> 
> Just to name a few.

Which job? Whatever you end up playing the most.

LankLTE commented:
> Which job? Whatever you end up playing the most.

Warden's my most played role, so I'll say that. I imagine if I filled my inventory with survival boxes I could fit the 3-4 things in that list I normally carry, but then I'm also giving up at least 1/7 inventory slots that could absolutely be used on something else better, like trying to grab contraband off of suspects I've detained or an item I'm bringing to the detective.

metalgearsloth commented:
> > Please list them off to me,
> 
> Sure, I'd be glad to. Cigarettes. Lighters. Drinks. Flashlights. Plushies. Figurines. PAI. Instruments. Soap. Books. Paper. Mice. Flowers. Silk. Hats. Suits. Jackets.
> 
> Just to name a few.

Gonna go out on a limb and say we shouldn't expect to be able to fit duplicate clothing on a person in regards to multiple spare suits / jackets, I'd expect just body armour on captain / hos / secoff or the likes.

LankLTE commented:
> Gonna go out on a limb and say we shouldn't expect to be able to fit duplicate clothing on a person in regards to multiple spare suits / jackets, I'd expect just body armour on captain / hos / secoff or the likes.

I know a lot of people like musicians / mines who enjoy being able to do that, but sure, knock those two off the list. The other 15 items are still things that’s perfectly valid to want to be carrying around without wasting one of their very limited item slots.

Eden077 commented:
Just from personal experience, when you're playing as a lizard having that extra coat on you at all times Is a lifeline for sci/Atmos fuck ups, very helpful for not immediately freezing to death cause you don't have a reliable HoP or the winterdrobe is out of stock

Nairodian commented:
> Which job? Whatever you end up playing the most.

I am a HoS main, my normal loadout goes a little like;
_**Basically Required:**_ flash, rubber mag, holo projector, glasses, door remote, space pen, epi pen, cloth/food, crowbar.
_**Can wear elsewhere but would rather keep in bag:**_ Oxy tank, oxy mask, combat gloves.
_**Handy but not required:**_ Body bag, stamp, pinpointer.
This is not including the times where things go extra lethal and I need to bring out my WT and a few spare mags because it eats lead like hell.

I generally want space to carry extra items like contraband dropped during a shootout, contraband I need to take to security to secure, items belonging to people in custody.
Or just generally personal effects like being able to take off my hat during high intense situations, gifts given to me, maybe a spare drink I just got from the bar.

Skarletto commented:
I guess boxes are meta now sheeesh
![image](https://github.com/space-wizards/space-station-14/assets/122584947/b36b5655-e665-4de8-8175-266b7e62a17b)

spectralSpices commented:
I feel like massively limiting the inventory like this will just make it harder to carry stuff you may just have sentimental attachment to. It'll encourage people to only grab shit that they'd consider essential, and would only really grant an advantage to people that know how/where to get that stuff immediately. I won't outright say "this'll encourage powergaming", because I don't believe that to entirely be the case, but I do think switching entirely to grid-based inventory is a bad idea.

keronshb commented:
> > Which job? Whatever you end up playing the most.
> 
> Warden's my most played role, so I'll say that. I imagine if I filled my inventory with survival boxes I could fit the 3-4 things in that list I normally carry, but then I'm also giving up at least 1/7 inventory slots that could absolutely be used on something else better, like trying to grab contraband off of suspects I've detained or an item I'm bringing to the detective.

You can still do all of that and more. Please download and test the PR for yourself to see.

keronshb commented:
> I guess boxes are meta now sheeesh ![image](https://user-images.githubusercontent.com/122584947/277553859-b36b5655-e665-4de8-8175-266b7e62a17b.png)

You don't even need boxes for that:

![277529237-2964a2fb-e5f7-4766-bb5a-1dc18aeea1f8](https://github.com/space-wizards/space-station-14/assets/54602815/e3bdf89a-8d72-42fc-bfc8-dba4ccc3a711)

keronshb commented:
> > Which job? Whatever you end up playing the most.
> 
> I am a HoS main, my normal loadout goes a little like; _**Basically Required:**_ flash, rubber mag, holo projector, glasses, door remote, space pen, epi pen, cloth/food, crowbar. _**Can wear elsewhere but would rather keep in bag:**_ Oxy tank, oxy mask, combat gloves. _**Handy but not required:**_ Body bag, stamp, pinpointer. This is not including the times where things go extra lethal and I need to bring out my WT and a few spare mags because it eats lead like hell.
> 
> I generally want space to carry extra items like contraband dropped during a shootout, contraband I need to take to security to secure, items belonging to people in custody. Or just generally personal effects like being able to take off my hat during high intense situations, gifts given to me, maybe a spare drink I just got from the bar.

![277549687-0064d971-522f-4d50-9973-2284520567f3](https://github.com/space-wizards/space-station-14/assets/54602815/ab260731-8572-4ed4-b007-69b5c19dc67a)

This is me lazily throwing together a powergaming setup (not even nearly optimized) and I can fit more than enough.

LankLTE commented:
> You don't even need boxes for that:
> 
> 
> 
> ![277529237-2964a2fb-e5f7-4766-bb5a-1dc18aeea1f8](https://github.com/space-wizards/space-station-14/assets/54602815/e3bdf89a-8d72-42fc-bfc8-dba4ccc3a711)

Mate you're using a box in the photo and are using up your *entire* storage with no room to spare.

metalgearsloth commented:
> > You don't even need boxes for that:
> > ![277529237-2964a2fb-e5f7-4766-bb5a-1dc18aeea1f8](https://user-images.githubusercontent.com/54602815/277555921-e3bdf89a-8d72-42fc-bfc8-dba4ccc3a711.png)
> 
> Mate you're using a box in the photo and are using up your _entire_ storage with no room to spare.

Ignoring the triple item count and capacity for more boxes tho

Skarletto commented:
> You don't even need boxes for that:
> ![277529237-2964a2fb-e5f7-4766-bb5a-1dc18aeea1f8](https://user-images.githubusercontent.com/54602815/277555921-e3bdf89a-8d72-42fc-bfc8-dba4ccc3a711.png)

what
you're literally using a box in that picture

Nairodian commented:
> ![277549687-0064d971-522f-4d50-9973-2284520567f3](https://user-images.githubusercontent.com/54602815/277556024-ab260731-8572-4ed4-b007-69b5c19dc67a.png)
> 
> This is me lazily throwing together a powergaming setup (not even nearly optimized) and I can fit more than enough.

Why would I want this. This looks like the opposite of what we want to achieve. Don't we want to AVOID making powergaming metas the norm? I swear we didn't WANT nested storage.

keronshb commented:
> > You don't even need boxes for that:
> > ![277529237-2964a2fb-e5f7-4766-bb5a-1dc18aeea1f8](https://user-images.githubusercontent.com/54602815/277555921-e3bdf89a-8d72-42fc-bfc8-dba4ccc3a711.png)
> 
> Mate you're using a box in the photo and are using up your _entire_ storage with no room to spare.

Can you not just jump down my throat?

Their setup was:

Flash, RCD, access config, door remote, stamp, x2 medipens, and a holofan projector.

You already start with x2 medipens on master and you can absolutely still put those items in a box on master.

The examples were to show that people are absolutely overblowing this storage limit out of proportion.

LankLTE commented:
Maybe this is a bold take but I'm not a huge fan of the meta being carrying 4 boxes on you at all times so that you can carry stuff you need. Isn't that essentially powergaming, what we're trying to avoid?

Eden077 commented:
>  > ![277549687-0064d971-522f-4d50-9973-2284520567f3](https://user-images.githubusercontent.com/54602815/277556024-ab260731-8572-4ed4-b007-69b5c19dc67a.png)
> > 
> > This is me lazily throwing together a powergaming setup (not even nearly optimized) and I can fit more than enough.
> 
> Why would I want this. This looks like the opposite of what we want to achieve. Don't we want to AVOID making powergaming metas the norm? I swear we didn't WANT nested storage.

Agreed, this seems more powergamey than what I can actually achieve with current storage

spectralSpices commented:
We are become Matryoshka Dolls, Destroyer Of Balance.

ElectroJr commented on Resources/Locale/en-US/components/storage-component.ftl:
```diff
@@ -1,9 +1,9 @@
 comp-storage-cant-drop = You can't let go of { THE($entity) }!
 comp-storage-window-title = Storage Item
-comp-storage-window-volume = Items: { $itemCount }, Stored: { $usedVolume }/{ $maxVolume }
-comp-storage-window-volume-unlimited = Items: { $itemCount }
+comp-storage-window-volume = Items: { $itemCount }/{ $maxCount }
```
If the storage has a max total weight, it should also show that ratio in the UI. Ideally it'd also show the weight of each individual item, but that'd require more significant UI changes, so it's probably fine to leave as is for now

Nairodian commented:
As I understand it. The new meta becomes go to cargo, get cardboard, make boxes. OR, steal boxes off of prisoners or dead bodies. As boxes are so fucking valuable now because you can hold six more items for each one you get.

metalgearsloth commented:
> As I understand it. The new meta becomes go to cargo, get cardboard, make boxes. OR, steal boxes off of prisoners or dead bodies. As boxes are so fucking valuable now because you can hold six more items for each one you get.

Of a lower weight class, not the same weight class.

LankLTE commented:
> Of a lower weight class, not the same weight class.

Things like pistols, magazines, meds, and tools, all of which you stated people hoarding was a problem, are in this lower weight class that people are now further encouraged to hoard because they can hold 7x as much as they can anything else

metalgearsloth commented:
> > Of a lower weight class, not the same weight class.
> 
> Things like pistols, magazines, meds, and tools, all of which you stated people hoarding was a problem, are in this lower weight class that people are now further encouraged to hoard because they can hold 7x as much as they can anything else

I tried to nerf meds and you also disagreed there https://github.com/space-wizards/space-station-14/pull/21117 so not really sure how problems are supposed to be fixed if you're against both solutions.

LankLTE commented:
> I tried to nerf meds and you also disagreed there https://github.com/space-wizards/space-station-14/pull/21117 so not really sure how problems are supposed to be fixed if you're against both solutions.

I disagreed with *tripling* the weight as I felt it was heavy handed, while also not fixing the alternatives people would just use instead. I'm also not against "both solutions", I'm against this as it's not a solution at all but actually allows people to hold significantly more medical supplies than they could before (49 stacks in one backpack if fit optimally compared to 10 before)

Eden077 commented:
> >  > ![277549687-0064d971-522f-4d50-9973-2284520567f3](https://user-images.githubusercontent.com/54602815/277556024-ab260731-8572-4ed4-b007-69b5c19dc67a.png)
> > > 
> > > This is me lazily throwing together a powergaming setup (not even nearly optimized) and I can fit more than enough.
> > 
> > Why would I want this. This looks like the opposite of what we want to achieve. Don't we want to AVOID making powergaming metas the norm? I swear we didn't WANT nested storage.
> 
> Agreed, this seems more powergamey than what I can actually achieve with current storage

I did some more analysis, um... Explain to me... How we would ever let a utility belt PLUS a sec belt in one bag... The powergaming that people will get away with is actually insane, you're calling "not even nearly optimized" when this is the most powergamey thing I've seen literally all year

ElectroJr commented on Resources/Locale/en-US/items/components/item-component.ftl:
```diff
@@ -7,3 +7,9 @@ pick-up-verb-get-data-text = Pick Up
 pick-up-verb-get-data-text-inventory = Put in hand
 
 item-component-on-examine-size = Size: {$size}
+
```
This examine text should also give the item's weight now that it also matters. I know weight is currently just based on size, but I guess that might change in the future and I don't think there's anything else to communicate that relationship with players.

metalgearsloth replied:
Backpack should show what size it fits for sure which is also much easier to communicate than a whitelist.

ElectroJr commented on Content.Shared/Item/ItemComponent.cs:
```diff
@@ -66,6 +65,38 @@ public VisualsChangedEvent(NetEntity item, string containerId)
+    /// </summary>
+    Normal = 3,
+
+    /// <summary>
+    /// Items that are too large to fit inside of standard bags, but can worn in exterior slots or placed in custom containers.
+    /// </summary>
+    Bulky = 4,
+
+    /// <summary>
+    /// Items that are too large to place inside of any kind of container.
+    /// </summary>
+    Large = 5
```
IMO it's not clear that "Large" is larger than "Bulky", I would've assumed the opposite. Maybe we should use Large & Huge instead?

metalgearsloth replied:
I think dnd uses Large -> Huge instead so might make more sense.

Also wouldn't it make more sense for the enum values to be the weight? Or alternatively use prototypes.

ElectroJr commented on Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:
```diff
@@ -589,28 +582,89 @@ public bool PlayerInsertHeldEntity(EntityUid uid, EntityUid player, StorageCompo
+        if (!Resolve(uid, ref uid.Comp))
+            return false;
+
+        return GetCumulativeItemSizes(uid, uid.Comp) < GetMaxTotalWeight(uid) &&
+               uid.Comp.Container.ContainedEntities.Count < uid.Comp.MaxSlots;
```
Invert order of checks

Aisu9 commented:
instead of nerfing medical items, why you don't nerf how hard it is to use them? like the brute and burn mender hand machine they use one goons?

JoeHammad1844 commented:
> I did some more analysis, um... Explain to me... How we would ever let a utility belt PLUS a sec belt in one bag... The powergaming that people will get away with is actually insane, you're calling "not even nearly optimized" when this is the most powergamey thing I've seen literally all year

if this hasnt been repeated, ss13 slot sizes go from tiny to small to normal to bulky, bulky being unable to be put in bags. just make belts and other non cardboard box type containers bulky

ElectroJr commented on Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:
```diff
@@ -589,28 +582,89 @@ public bool PlayerInsertHeldEntity(EntityUid uid, EntityUid player, StorageCompo
+        return (ItemSize) Math.Max((int) item.Size - 1, 1);
+    }
+
+    public int GetMaxTotalWeight(Entity<StorageComponent?> uid)
+    {
+        if (!Resolve(uid, ref uid.Comp))
+            return 0;
+
+        if (uid.Comp.MaxTotalWeight != null)
+            return uid.Comp.MaxTotalWeight.Value;
+
+        return uid.Comp.MaxSlots * SharedItemSystem.GetItemSizeWeight(GetMaxItemSize(uid));
+    }
```
AFAICT, if the max weight is not explicitly defined, then it will never be a limiting factor - the max item count will always block insertions.

Unless there'd ever be a situation where we'd want the automatic max total weight to be more restrictive, then all of the code that calls `GetMaxTotalWeight()` should just get wrapped in a `if (comp.MaxTotalWeight != null)` instead, as that'd avoid having to constantly compute the current total weight, which itself requires many component lookups.

ElectroJr replied:
Or is the plan to eventually allow things to modify weight (i.e., the weight of a bag includes all contents). In which case this'd no longer be the case and you'd still have to always compute the total weight.

metalgearsloth replied:
At least in ss13 count can be a limiting factor if you put a lot of small stuff in though for now I think the goal is to get the system first and balance more after as changing all of the yaml is a large piece of work.

EmoGarbage404 replied:
it's an optional balancing factor that storages can opt to use.

ElectroJr reviewed (dismissed):
This is not at all a thorough review, just some things I noticed.

JoeHammad1844 commented:
also, keep in mind that ss13 uses sutures and regen mesh as their common topicals, our common topicals are literally half as good

Aisu9 commented:
> also, keep in mind that ss13 uses sutures and regen mesh as their common topicals, our common topicals are literally half as good

if anything the bruise packs and ointment should be removed from medbay in favor for the menders from goons, and advanced medical items because they were meant to be craftable ONLY and not round start and in every fucking medkit

JoeHammad1844 commented:
the way ss13 does slot storage is good

CrigCrag commented:
this is incredibly unintuitive, why should a backpack only be able to hold more small items if it has a smaller container inside it?
here we are, 500+ years in the future, with backpacks that can only hold 7 pills before suddenly being too full to fit a single sheet of paper.

JoeHammad1844 commented:
im just happy that survival boxes arent useless anymore, and hopefully medkits can be stored in bags

LookyPadooky commented:
I will pay you to close this PR

metalgearsloth commented:
> I will pay you to close this PR

Please don't post garbage comments

LookyPadooky commented:
> > I will pay you to close this PR
> 
> Please don't post garbage comments

Genuinely I must know what this adds besides being more like 13, I read emos whole argument and none of the pros even seem like they would matter. Also I was under the impression all of you hated powergamers and metagamers this system seems made to ONLY help them. the last thing i need in this game is to have 5 inventory windows open at all times. This just seems like a ton of wasted effort because I know it will get reverted in less than a day when the masses that don't read github find out it got added. Please do not backtrack on good ideas to be more like a 20 year old game when there's a new engine and a fresh start for a reason.

Tunguso4ka commented:
i would like to see ss13 style inventory UI too. SS14 kinda looks lame.

deltanedas commented:
this is all spiraling off muh sosa holding 100 bruise packs in his bag
so, just make it so you cant self heal or its 10x slower than it already is
then med isnt affected in any way since you are healing other people and sosa cant use all those meds, win win

then instead of holding 490 bruise packs in bag in new ultra deluxe powergamer system which you can still use to self heal in space, just fill your bag exclusively to heal other people.

metalgearsloth commented on Content.IntegrationTests/Tests/StorageTest.cs:
```diff
@@ -32,9 +32,11 @@ public async Task StorageSizeArbitrageTest()
                         storage.Whitelist != null ||
-                        !proto.TryGetComponent<ItemComponent>("Item", out var item)) continue;
+                        storage.MaxItemSize == null ||
+                        !proto.TryGetComponent<ItemComponent>("Item", out var item))
+                        continue;
```
Would be easier to just use compfactory to grab the compname but not gonna nitpick the pr over it.

ZeroDayDaemon commented:
I am not against this change, It falls more in line with Space Station 13, I understand people who have never/barely played SS13 not liking it, but I feel like slot storage is actually much more simple.

JoeHammad1844 commented:
before criticizing this PR, i invite you all to play SS13 (TG, fulp, goon, paradise) for a good amount of time, to understand how it feels in practice

Endecc commented:
> I am not against this change, It falls more in line with Space Station 13, I understand people who have never/barely played SS13 not liking it, but I feel like slot storage is actually much more simple.

Having had my fair share of time with SS13 as well, I strongly disagree that slot storage is "much more simple". With such a reliance on nested storage, it brings up a lot more UI fumbling, or having 3 to 5 inventories open at once to have ready access to what you need. This isn't intuitive or fun, **and** is downright obtrusive to anyone with a lower resolution monitor.

And it falling more in line with SS13 is not a valid reason for any change. We have more accessible systems, like inventory management, that don't need "realignment" to the "old ways". SS14 wasn't remade to be a mirror copy of the SS13 but in a new engine, nor should it ever be.

UbaserB commented:
> before criticizing this PR, i invite you all to play SS13 (TG, fulp, goon, paradise) for a good amount of time, to understand how it feels in practice

i played it for a bit and actually didn't mind it at all. It wasn't too bad.

metalgearsloth commented:
> Having had my fair share of time with SS13 as well, I strongly disagree that slot storage is "much more simple". With such a reliance on nested storage, it brings up a lot more UI fumbling, or having 3 to 5 inventories open at once to have ready access to what you need. This isn't intuitive or fun, and is downright obtrusive to anyone with a lower resolution monitor.

Limiting storage windows would favour lower resolution monitors I would think, no? At the moment higher-resolution monitors are favoured on master. But that's also a separate PR so I digress.

On the med PR you made this comment:

> This balance change is gutting far more appropriately tuned aspects that don't need adjustment, than it is nerfing what it needs to. If there's a better solution, find it.

The solution has been found but I welcome if you have any specific suggestions on fixing issues.

metalgearsloth commented on Content.Shared/Item/SharedItemSystem.cs:
```diff
@@ -83,14 +86,7 @@ private void OnHandInteract(EntityUid uid, ItemComponent component, InteractHand
 
-        if (!_prototype.TryIndex<StackPrototype>(stack.StackTypeId, out var stackProto) ||
-            stackProto.ItemSize is not { } size)
-            return;
-
-        SetSize(uid, args.NewCount * size, component);
     }
```
Is this a dummy method now?

EmoGarbage404 replied:
there's an inherited method on the client

metalgearsloth replied:
whacky

metalgearsloth commented on Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:
```diff
@@ -589,28 +582,89 @@ public bool PlayerInsertHeldEntity(EntityUid uid, EntityUid player, StorageCompo
-    public bool PlayerInsertEntityInWorld(EntityUid uid, EntityUid player, EntityUid toInsert, StorageComponent? storageComp = null)
+    public bool PlayerInsertEntityInWorld(Entity<StorageComponent?> uid, EntityUid player, EntityUid toInsert)
     {
-        if (!Resolve(uid, ref storageComp) || !_sharedInteractionSystem.InRangeUnobstructed(player, uid))
+        if (!Resolve(uid, ref uid.Comp) || !_sharedInteractionSystem.InRangeUnobstructed(player, uid))
```
Do we note have a Resolve(ref uid) yet? Probably easier to just add it.

deltanedas replied:
its missing also it would have to account for taking in Entity<XComponent> instead of Entity<XComponent?>

metalgearsloth commented on Content.Shared/Storage/StorageComponent.cs:
```diff
@@ -20,6 +22,25 @@ public sealed partial class StorageComponent : Component
+        [DataField, ViewVariables(VVAccess.ReadWrite), AutoNetworkedField]
+        public int MaxSlots = 7;
+
+        /// <summary>
+        /// The maximum size item that can be inserted into this storage,
+        /// </summary>
+        [DataField, ViewVariables(VVAccess.ReadWrite), AutoNetworkedField]
+        [Access(typeof(SharedStorageSystem))]
+        public ItemSize? MaxItemSize;
```
Might be safest to just use normal? not sure.

EmoGarbage404 replied:
a null value is interpreted as "one less than the entity's item size" This will dodge a lot of test fails that could happen easily otherwise.

Endecc commented:
> Limiting storage windows would favour lower resolution monitors I would think, no? At the moment higher-resolution monitors are favoured on master. But that's also a separate PR so I digress.

If this is calling back to your PR #20335, then this still creates more UI fumbling, **especially more so** when encouraging box hoarding like _this_ PR does. It's just adding tedium on top of tedium and doesn't make for any convenience or improvement. It's making things annoying for the sake of being annoying, which is not helpful for balance or usability.

> The solution has been found but I welcome if you have any specific suggestions on fixing issues.

No, it has not. In fact I would go to say this "solution" to topical hoarding is gutting **even more** appropriately tuned aspects across **the entire game**, because there is supposedly an issue with people becoming lone armies.

My suggestion, after absorbing the discussions across your weight increase PR and this one;
Decrease self-treatment speed using topicals by 50%. Maybe you could even cut the self-treated healing amount, so it has an explicit use case of being for emergencies and getting you to someone who can actually help. Is that acceptable?

metalgearsloth commented:
> My suggestion, after absorbing the discussions across your weight increase PR and this one;
Decrease self-treatment speed using topicals by 50%.

This doesn't address the myriad of other issues raised with storage that were raised.

- Standardising item sizes even if we don't take slots is very good.
- Removing manually curating whitelists is very very good.
- Removing 9999 capacity memes is good.

So how about we just try it the ss13, selectively reverting is easy enough if we end up not liking it.

Endecc commented:
> This doesn't address the myriad of other issues raised with storage that were raised.

You quoted my comment on the Topicals PR, asked for my suggestion on it, and I gave it as you wanted.
Several other people have contributed to the overarching balance discussion, and I'm not about to reiterate the same things they've said for the 4th time. Please read @LankLTE, @Eden077, and @Nairodian's comments.

CatMagic1 commented:
I think there are pros to both systems, **and** I also think it would be a better use of time and effort to come up with a system that improves on the cons that have been presented for both systems. We could also just go back to passive-aggressively fighting, and just keep making bigger and more unrelated prs that all get closed because the same people argue against them and in the end, solve nothing as a community. 👍

metalgearsloth commented:
> making bigger and more unrelated prs that all get closed because the same people argue against them and in the end, solve nothing as a community. 

On the contrary the code cleanup here is invaluable and it's getting merged, barring the game exploding of course.

arimah commented:
I don't have an enormous amount to add to this discussion, as others have already raised my criticisms for me, but I will comment on this:

> So how about we just try it the ss13 way

This PR _fundamentally changes_ one of the most basic interactions and most used features in the entire game, around which a great deal of balacing has already been performed. As has been pointed out by others, it will have **drastic** consequences on gameplay for thousands (tens of thousands?) of players, and on downstream for the _many_ forks that exist. I feel these radical overhauls really should not be performed in so gung-ho a fashion, but should be the result of a long discussion and broad consensus – not just within the maintainer sphere but among players also.

CatMagic1 commented:
> > making bigger and more unrelated prs that all get closed because the same people argue against them and in the end, solve nothing as a community.
> 
> On the contrary the code cleanup here is invaluable and it's getting merged, barring the game exploding of course.

As long as we don't have slot-based storage next time I log into leviathan.

metalgearsloth commented:
> We could also just go back to passive-aggressively fighting, and just keep making bigger and more unrelated prs that all get closed because the same people argue against them and in the end, solve nothing as a community.

> As long as we don't have slot-based storage next time I log into leviathan.

Be civil or refrain from commenting please.

GoodWheatley commented:
> On the contrary the code cleanup here is invaluable and it's getting merged, barring the game exploding of course.

If the code cleanup is important enough to be merged in of itself, then why isn’t this PR atomized? In its current state all the dissent on this PR is practically irrelevant since it’ll be merged regardless for code maintenance, which from my perspective feels intentional.

metalgearsloth commented:
Also had this localisation error locally:

![image](https://github.com/space-wizards/space-station-14/assets/31366439/dbe2f538-ab41-45cd-ab83-8846fd053e6d)

EmoGarbage404 commented:
> Also had this localisation error locally:
> 
> ![image](https://user-images.githubusercontent.com/31366439/277894950-dbe2f538-ab41-45cd-ab83-8846fd053e6d.png)

fixed and added max size
![image](https://github.com/space-wizards/space-station-14/assets/98561806/9fde1394-4cc7-4007-a0f9-6e9ca13c1407)

Delams-The-SM commented:
Trying to wrap my head about all those lines of changed code but not all comments seem all that . . . useful
![image](https://github.com/space-wizards/space-station-14/assets/66973657/81f03ed2-d4b9-405e-8615-deb457b41738)

EmoGarbage404 commented:
![image](https://github.com/space-wizards/space-station-14/assets/98561806/095e28a6-4b54-4653-88ea-93e3cb02f32b)

added UI feedback for weight limit.

AJCM-git commented:
It has been decided to test merge this and see what effect it has in the game. We currently don't have an automated test-merge system so we will do it the usual manual way where if the change generates unforeseen problems during testing we will revert it (of course if the changes work as intended they will stay).

NancokPS2 commented:
3:1 ratio of denial over aproval. Clear outlined issues that barely fix the existing "problems" and cause many more, gets test merged anyway.  
Why care about what players find fun when you can make the system more optimal, right?

metalgearsloth commented:
> 3:1 ratio of denial over aproval. Clear outlined issues that barely fix the existing "problems" and cause many more, gets test merged anyway. Why care about what players find fun when you can make the system more optimal, right?

Take your vitriol elsewhere and if you make another comment like this on github I won't hesitate to ban.

NancokPS2 commented:
Yeah, i can tell how the attitude towards opinions is so far (as in, nobody cares). I already installed BYOND, i'll be back in a few years if anything.

EmoGarbage404 commented:
> Yeah, i can tell how the attitude towards opinions is so far (as in, nobody cares). I already installed BYOND, i'll be back in a few years if anything.

let us know how it goes considering it's the ss13 inventory system.

CatMagic1 commented:
> Take your vitriol elsewhere and if you make another comment like this on github I won't hesitate to ban.

ackin like they aren't telling the truth

stop saying people are being overly rude when you do the same thing

Carolyn3114 commented:
> let us know how it goes considering it's the ss13 inventory system.

this still isn't how it works in ss13

Us3rn4me228 commented:
> ## About the PR
> Fixes #21134
> 
> Storage now functions on a slot-based system instead of a numerical size one.
> 
> Storages now specify an amount of items and an optional max size and optional max total weight. Leaving the max size or max total weight null simply fills it in with appropriate defaults.
> 
> Sizes are defined with an enum, and carry an additional "weight cost" which is just used internally to stop players from stuffing bags with tons of bulky items.
> 
> ## Why / Balance
> all of it is in the issue.
> 
> Most of my argument is articulated here: [#21134 (comment)](https://github.com/space-wizards/space-station-14/issues/21134#issuecomment-1773808371)
> 
> ## Technical details
> ItemComponent's Size is now an enum rather than an int.
> 
> In StorageComponent, Capacity has been replaced with MaxSlots, MaxItemSize (nullable) and MaxTotalWeight (nullable)
> 
> ## Media
> ![image](https://user-images.githubusercontent.com/98561806/277525507-486f1869-56ce-4cab-a99a-321846ee1767.png) ![image](https://user-images.githubusercontent.com/98561806/277525544-bd09aa42-2842-4b5d-b572-ee9c9d89a396.png)
> 
> * [x]  I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
> 
> ## Breaking changes
> Some ItemSystem functions take in the ItemSize enum rather than a simple int.
> 
> A few storageSystem functions now use Entity arguments in the header.
> 
> Removed Capacity, UsedStorage vars from StorageComponent.
> 
> **Changelog**
> 
> 🆑
> 
> * tweak: Storage now functions based on slots and sizes, rather than numerical values.

DrSmugleaf commented:
> ackin like they aren't telling the truth
> 
> stop saying people are being overly rude when you do the same thing

We have a section on the forums where you can leave a staff complaint if you feel that one of them is being rude.
Conversely, next time you leave a needlessly confrontational and off-topic comment like this on GitHub you are getting repo-banned. You were already warned in this same thread, let this be your last warning.

Us3rn4me228 commented:
Making a slot inventory system is the most stupid and idiotic decision that could have ever been made.
SS13 had such an inventory system because of the limitations of the BYOND engine.
And in SS14 you've already made an inventory system that is MANY times better than what was there before.
SO WHY ARE YOU GUYS BREAKING SOMETHING THAT JUST ALREADY WORKING FINE????
WHAT GENUINELY WAS THE THOUGHT PROCCES HERE?

sTiKyt commented:
> > ackin like they aren't telling the truth
> > 
> > stop saying people are being overly rude when you do the same thing
> 
> We have a section on the forums where you can leave a staff complaint if you feel that one of them is being rude.
> Conversely, next time you leave a needlessly confrontational and off-topic comment like this on GitHub you are getting repo-banned. You were already warned in this same thread, let this be your last warning.

Ah yeah, that section.. Section that gets you banned and bullied by staff, lmao

CatMagic1 commented:
> We have a section on the forums where you can leave a staff complaint if you feel that one of them is being rude. Conversely, next time you leave a needlessly confrontational and off-topic comment like this on GitHub you are getting repo-banned. You were already warned in this same thread, let this be your last warning.

Sorry for being rude, but you CAN see why people would be mad right? I just type too fast to realize I shouldn't say things... Anyway, you said I was warned in this thread but I really don't know what I was warned for?? Maybe the "As long as we don't have slot-based storage next time I log into leviathan." but I really only said that because I wasn't sure if he meant they were merging the slot-based storage or just the optimizations, because y'know (It seemed unlikely that they would merge it considering the mass opposition for the change, I mean it's only test merged which is still a little weird just because of how MANY people said they didn't want this). I'm optimistic though and I think I'll check it out later.

_mark this as off-topic_

IProduceWidgets commented:
After playing a few rounds with it, I don't mind the size standardization (though I can see it will need tweaks mostly to do with the large vs normal distinction). I do not like the (practically) forced inventory nesting. I'm okay with some containers providing increased space. I can understand that as maybe pill bottle are more organized or similar. However, I think there needs to be a way to put multiple smaller objects into larger slots as has been discussed a bit.
